crates/turborepo-lib/src/run/mod.rs

```diff
@@ -364,8 +366,30 @@ impl<'a> Run<'a> {
 
         let scm = SCM::new(&self.base.repo_root);
 
-        let filtered_pkgs =
-            scope::resolve_packages(&opts.scope_opts, &self.base.repo_root, &pkg_dep_graph, &scm)?;
+        let filtered_pkgs = {
+            let mut filtered_pkgs = scope::resolve_packages(
+                &opts.scope_opts,
+                &self.base.repo_root,
+                &pkg_dep_graph,
+                &scm,
+            )?;
+
+            if filtered_pkgs.len() != pkg_dep_graph.len() {
+                for target in self.targets() {
+                    let task_name = TaskName {
+                        package: Some(ROOT_PKG_NAME.into()),
+                        task: target.into(),
+                    };
```

Comment:
I don't think this is correct, if a user passes `//#build` as a target then the task name this constructs is `TaskName { package: Some("//"), task: "//#build" }`

Reply:
Can you even do that? I don't think `//#build` is a valid target:

Reply:
Workspace tasks, including root tasks, are only valid if they're defined in `turbo.json` so `//#build` is only valid if it appears in the pipeline e.g. we can run `turbo run //#build:ts` in our monorepo since we define ` //#build:ts` pipeline, but we can't run a `//#build` since it isn't present (or `docs#build` for that matter).